lib/matplotlib/patches.py

```diff
@@ -1619,12 +1617,14 @@ def iter_circle_intersect_on_line(x0, y0, x1, y1):
                 else:
                     sign_dy = 1.0
                 sqrt_discrim = np.sqrt(discrim)
-                for sign in (1., -1.):
-                    x = (D * dy + sign * sign_dy * dx * sqrt_discrim) / dr2
-                    y = (-D * dx + sign * np.abs(dy) * sqrt_discrim) / dr2
-                    yield x, y
+                return np.row_stack(
```

Comment:
Don't know. Better to write the 2x2 array out in all components?

Reply:
would be quite a mouthful, no?

Reply:
A loop to define an array always twists my brain :smile: Don't know what's better:

~~~
return np.row_stack(
    [[(D * dy + sign * sign_dy * dx * sqrt_discrim) / dr2,
      (-D * dx + sign * np.abs(dy) * sqrt_discrim) / dr2]
     for sign in [1, -1]])
~~~
or
~~~
return np.array(
    [[D * dy + sign_dy * dx * sqrt_discrim / dr2,
      -D * dx + np.abs(dy) * sqrt_discrim / dr2],
     [D * dy - sign_dy * dx * sqrt_discrim / dr2,
      -D * dx - np.abs(dy) * sqrt_discrim / dr2]])
~~~
or
~~~
return (D * np.array([[dy, -dx],
                      [dy, -dx]])
        + sqrt_discrim / dr2 * np.array([[sign_dy * dx, np.abs(dy)]
                                         [-sign_dy * dx, -np.abs(dy)]]))
~~~

Anyway, I'll approve and you decide which way to go.